src/app/stepprocess/stepprocess-auto-step-change.demo.ts

```diff
@@ -36,70 +38,23 @@ export class StepProcessAutoStepChangeComponent implements OnInit, AfterViewInit
-        substeps: [
-          { id: 'step5Sub1', title: 'Step 5 Sub 1', icon: 'success', content: 'Step 5 Sub 1 content' }, // eslint-disable-line
-          { id: 'step5Sub2', title: 'Step 5 Sub 2', icon: 'empty-circle', content: 'Step 5 Sub 2 content' }, // eslint-disable-line
-          { id: 'step5Sub3', title: 'Step 5 Sub 3', icon: 'error', content: 'Step 5 Sub 3 content' }, // eslint-disable-line
-          { id: 'step5Sub4', title: 'Step 5 Sub 4', icon: 'empty-circle', content: 'Step 5 Sub 4 content' }, // eslint-disable-line
-          { id: 'step5Sub5', title: 'Step 5 Sub 5', icon: 'empty-circle', content: 'Step 5 Sub 5 content' }, // eslint-disable-line
-          { id: 'step5Sub6', title: 'Step 5 Sub 6', icon: 'error', content: 'Step 5 Sub 6 content' }, // eslint-disable-line
-        ]
-      }, // eslint-disable-line
-      { id: 'step6', title: 'Step 6', icon: 'empty-circle', content: 'Step 6 content' }, // eslint-disable-line
-      { id: 'step7', title: 'Step 7', icon: 'error', content: 'Step 7 content' }, // eslint-disable-line
-      { id: 'step8', title: 'Step 8', icon: 'success', content: 'Step 8 content' }, // eslint-disable-line
-      { id: 'step9', title: 'Step 9', icon: 'success', content: 'Step 9 content' }, // eslint-disable-line
-    ];
+      }
+    ]
   }
 
-  ngAfterViewInit() {
-
-    setTimeout(() => {
-      if (this.stepListItems) {
-        const stepListItemsLength = this.stepListItems.toArray().length;
-        const stepNo = Math.floor(Math.random() * stepListItemsLength);
-        this.stepListItems.toArray()[stepNo].isSelected = true;
-        console.log(stepNo);
-        console.log('step changed');
-      }
-    }, 5000);
-    setTimeout(() => {
-      if (this.stepListItems) {
-        const stepListItemsLength = this.stepListItems.toArray().length;
-        const stepNo = Math.floor(Math.random() * stepListItemsLength);
-        this.stepListItems.toArray()[stepNo].isSelected = true;
-        console.log(stepNo);
-        console.log('step changed');
-      }
-    }, 7000);
-    setTimeout(() => {
-      if (this.stepListItems) {
-        const stepListItemsLength = this.stepListItems.toArray().length;
-        const stepNo = Math.floor(Math.random() * stepListItemsLength);
-        this.stepListItems.toArray()[stepNo].isSelected = true;
-        console.log(stepNo);
-        console.log('step changed');
-      }
-    }, 9000);
-    setTimeout(() => {
-      if (this.stepListItems) {
-        const stepListItemsLength = this.stepListItems.toArray().length;
-        const stepNo = Math.floor(Math.random() * stepListItemsLength);
-        this.stepListItems.toArray()[stepNo].isSelected = true;
-        console.log(stepNo);
-        console.log('step changed');
-      }
-    }, 11000);
 
+  selectPanel(id: string) {
+    console.log(id);
+    const selectedItem = this.stepListItems.find((item) => item.stepId === id);
+    if (selectedItem) {
+      selectedItem.isSelected = true;
+    }
```

Comment:
and here your could just:
```suggestion
    this.selectedStepId = id;
```